pkg/rtcp/raw_packet.go

```diff
@@ -0,0 +1,34 @@
+package rtcp
+
+// RawPacket represents an unparsed RTCP packet. It's returned by Unmarshal when
+// a packet with an unknown type is encountered.
+type RawPacket []byte
+
+// Marshal encodes the packet in binary.
+func (r RawPacket) Marshal() ([]byte, error) {
+	return r, nil
+}
+
+// Unmarshal decodes the packet from binary.
+func (r *RawPacket) Unmarshal(b []byte) error {
+	if len(b) < (headerLength) {
+		return errPacketTooShort
+	}
+	*r = b
+
+	var h Header
+	if err := h.Unmarshal(b); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Header returns the Header associated with this packet.
+func (r RawPacket) Header() Header {
+	var h Header
+	if err := h.Unmarshal(r); err != nil {
+		return Header{}
```

Comment:
What's a clear way to check if this is a valid header? can we add a `func (h Header) valid() bool` somewhere?

Reply:
Yeah I don't like eating that Unmarshal error here, but I don't think it makes sense to return an error from Header either.

You'll find out the packet is corrupted when you try to unmarshal it anyway so I think it's OK? The header.Type being 0 will also give a clue.